pkg/reconciler/ingress/ingress.go

```diff
@@ -192,14 +205,29 @@ func markIngressReady(ingress *v1alpha1.Ingress) {
 	ingress.Status.MarkNetworkConfigured()
 }
 
-// TODO(bvennam) we need to pull this from the upstream ingress that is create "-new"
 func domainForLocalGateway(ingressName string, isPrivate bool) string {
+	// checks for a valid domain in the list of load balancers
+	if LBDomain, ok := loadBalancers[strings.Split(ingressName, ".")[0]]; ok {
```

Comment:
What if not "ok"? Need a check

Reply:
If not "ok" this uses the default load balancer domain.

Added testing for malformed entries, this now accepts only istio/kourier as valid ingress and load balancer domains.